lib/rubocop/cop/minitest/assert_nil.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Minitest
+      # Check if your test uses `assert_nil` instead of `assert_equal(nil, something)`.
+      #
+      # @example
+      #   # bad
+      #   assert_equal(nil, actual)
+      #   assert_equal(nil, actual, 'the message')
+      #
+      #   # good
+      #   assert_nil(actual)
+      #   assert_nil(actual, 'the message')
+      #
+      class AssertNil < Cop
+        MSG = 'Prefer using `assert_nil(%<arguments>s)` over ' \
+              '`assert_equal(nil, %<arguments>s)`.'
+
+        def_node_matcher :assert_equal_with_nil, <<~PATTERN
+          (send nil? :assert_equal nil $_ $...)
+        PATTERN
+
+        def on_send(node)
+          assert_equal_with_nil(node) do |argument, assertion_message|
```

Comment:
The matched pattern may be concrete if block variable names are as follows. 
e.g: 

```diff
-assert_equal_with_nil(node) do |argument, assertion_message|
+assert_equal_with_nil(node) do |actual, message|
```